langchain/src/chat_models/openai.ts

```diff
@@ -712,56 +589,83 @@ export class ChatOpenAI
     return { totalCount, countPerMessage };
   }
 
-  /** @ignore */
+  /**
+   * Calls the OpenAI API with retry logic in case of failures.
+   * @param request The request to send to the OpenAI API.
+   * @param options Optional configuration for the API call.
+   * @returns The response from the OpenAI API.
+   */
+  async streamingCompletionWithRetry(
+    request: OpenAIClient.Chat.CompletionCreateParamsStreaming,
+    options?: OpenAICoreRequestOptions
+  ): Promise<AsyncIterable<OpenAIClient.Chat.Completions.ChatCompletionChunk>> {
+    const requestOptions = this._getClientOptions(options);
+    const fn: (
+      body: OpenAIClient.Chat.CompletionCreateParamsStreaming,
+      options?: OpenAICoreRequestOptions
+    ) => Promise<
+      AsyncIterable<OpenAIClient.Chat.Completions.ChatCompletionChunk>
+    > = this.client.chat.completions.create.bind(this.client);
+    return this.caller.call(fn, request, requestOptions).then((res) => res);
+  }
+
+  /**
+   * Calls the OpenAI API with retry logic in case of failures.
+   * @param request The request to send to the OpenAI API.
+   * @param options Optional configuration for the API call.
+   * @returns The response from the OpenAI API.
+   */
   async completionWithRetry(
-    request: CreateChatCompletionRequest,
-    options?: StreamingAxiosConfiguration
-  ) {
+    request: OpenAIClient.Chat.CompletionCreateParamsNonStreaming,
+    options?: OpenAICoreRequestOptions
+  ): Promise<OpenAIClient.Chat.Completions.ChatCompletion> {
+    const requestOptions = this._getClientOptions(options);
+    const fn: (
+      body: OpenAIClient.Chat.CompletionCreateParamsNonStreaming,
+      options?: OpenAICoreRequestOptions
+    ) => Promise<OpenAIClient.Chat.Completions.ChatCompletion> =
+      this.client.chat.completions.create.bind(this.client);
+    return this.caller.call(fn, request, requestOptions).then((res) => res);
+  }
+
+  private _getClientOptions(options: OpenAICoreRequestOptions | undefined) {
     if (!this.client) {
       const openAIEndpointConfig: OpenAIEndpointConfig = {
         azureOpenAIApiDeploymentName: this.azureOpenAIApiDeploymentName,
         azureOpenAIApiInstanceName: this.azureOpenAIApiInstanceName,
         azureOpenAIApiKey: this.azureOpenAIApiKey,
         azureOpenAIBasePath: this.azureOpenAIBasePath,
-        basePath: this.clientConfig.basePath,
+        baseURL: this.clientConfig.baseURL,
       };
 
       const endpoint = getEndpoint(openAIEndpointConfig);
-
-      const clientConfig = new Configuration({
+      const params = {
         ...this.clientConfig,
-        basePath: endpoint,
-        baseOptions: {
-          timeout: this.timeout,
-          ...this.clientConfig.baseOptions,
-        },
-      });
+        baseURL: endpoint,
+        timeout: this.timeout,
+        maxRetries: 0,
+      };
+      if (!params.baseURL) {
```

Comment:
@rattrayalex would be nice to not have to do this `delete` - could `baseURL: undefined` default to just the default OAI setting?

Reply:
Ah, sorry about this - agree that's a bug, I've filed a ticket to fix.